integration/options.go

```diff
@@ -0,0 +1,142 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"time"
+)
+
+const (
+	// defaultServerStateChangeTimeout is the default time we wait for a server to change its state.
+	defaultServerStateChangeTimeout = 30 * time.Second
+
+	// defaultReadRequestTimeout is the default read request timeout.
+	defaultReadRequestTimeout = 5 * time.Second
+
+	// defaultWriteRequestTimeout is the default write request timeout.
+	defaultWriteRequestTimeout = 5 * time.Second
+
+	// defaultUseTChannelClientForReading determines whether we use the tchannel client for reading by default.
+	defaultUseTChannelClientForReading = true
+
+	// defaultUseTChannelClientForWriting determines whether we use the tchannel client for writing by default.
+	defaultUseTChannelClientForWriting = false
+)
+
+type testOptions interface {
+	// ServerStateChangeTimeout sets the server state change timeout.
+	ServerStateChangeTimeout(value time.Duration) testOptions
+
+	// GetServerStateChangeTimeout returns the server state change timeout.
+	GetServerStateChangeTimeout() time.Duration
+
+	// ReadRequestTimeout sets the read request timeout.
+	ReadRequestTimeout(value time.Duration) testOptions
+
+	// GetReadRequestTimeout returns the read request timeout.
+	GetReadRequestTimeout() time.Duration
+
+	// WriteRequestTimeout sets the write request timeout.
+	WriteRequestTimeout(value time.Duration) testOptions
+
+	// GetWriteRequestTimeout returns the write request timeout.
+	GetWriteRequestTimeout() time.Duration
+
+	// UseTChannelClientForReading sets whether we use the tchannel client for reading.
+	UseTChannelClientForReading(value bool) testOptions
+
+	// GetUseTChannelClientForReading returns whether we use the tchannel client for reading.
+	GetUseTChannelClientForReading() bool
+
+	// UseTChannelClientForWriting sets whether we use the tchannel client for writing.
+	UseTChannelClientForWriting(value bool) testOptions
+
+	// GetUseTChannelClientForWriting returns whether we use the tchannel client for writing.
+	GetUseTChannelClientForWriting() bool
+}
+
+type options struct {
+	serverStateChangeTimeout    time.Duration
+	readRequestTimeout          time.Duration
+	writeRequestTimeout         time.Duration
+	useTChannelClientForReading bool
+	useTChannelClientForWriting bool
+}
+
+func newOptions() testOptions {
```

Comment:
newTestOptions for consistency perhaps?  On my current diff im changing the tests that aren't prefixed with new to include it for creating options.